pxfish/field_type.py

```diff
@@ -0,0 +1,145 @@
+            definition=field_type_definition,
+            role='input', session=session)
+        )
+
+    for field_type_definition in definitions['outputs']:
+        field_types.append(add_field_type(
+            operation_type=operation_type,
+            definition=field_type_definition,
+            role='output', session=session)
+        )
+
+    operation_type.field_types = field_types
+
+    session.utils.update_operation_type(operation_type)
+
+
+def add_aft(*, session, definition):
+    """
+    Adds Sample and Object type names to Field Type Object
+
+    Arguments:
+        session (Session Object): Aquarium session object
+        definition (Dictionary): Data about Sample and Object types
+    """
+    afts = []
+    for aft in definition['allowable_field_types']:
+
+        sample_type = session.SampleType.new()
+        object_type = session.ObjectType.new()
+
+        sample_type.name = aft['sample_type']
+        object_type.name = aft['object_type']
+        afts.append({'sample_type': sample_type, 'object_type': object_type})
+    return afts
+
+
+def equivalent(*, field_type, definition):
+    """
+    Compares a field type object with a definition.
+    Returns True if equivalent, False otherwise.
+    """
+    return True
+
+
+def types_valid(*, operation_type, definitions, session):
+    """
+    Compares an Operation Type's Field Types to Those in the definitions file
+    If a Field Type only exists in the Instance Type, returns False
+    Arguments:
+        operation_type (Operation Type): parent object
+        definitions (Dictionary): data about field types
+        session (Session Object): Aquarium session object
+    """
+    field_types = session.FieldType.where({'parent_id': operation_type.id})
+    type_map = {field_type.name: field_type for field_type in field_types}
+
+    local_diff = dict()
+    match_names = set()
+    conflicts = dict()
+    for definition in definitions['inputs']:
+        name = definition['name']
+        if name in type_map:
+            match_names.add(name)
+            if not equivalent(field_type=type_map[name], definition=definition):
+                conflicts[name] = definition
+        else:
+            local_diff[name] = definition
+
+    for definition in definitions['outputs']:
+        name = definition['name']
+        if name in type_map:
+            match_names.add(name)
+            if not equivalent(field_type=type_map[name], definition=definition):
+                conflicts[name] = definition
+        else:
+            local_diff[name] = definition
+
+    aquarium_diff_names = type_map.keys().difference(match_names)
+
+    # match_names: names that occur in both
+    # conflicts: dictionary of definitions not equivalent to aq field type with same name
+    # local_diff: dictionary of definitions with no aq field type with same name
+    # aquarium_diff_names: aq field type names that have no matching definition
+
+    if aquarium_diff_names:
```

Comment:
Please change logging so that it uses the captured differences to list the conflicts in the message